lib/rubycritic/platforms/base.rb

```diff
@@ -0,0 +1,43 @@
+require "rubycritic/configuration"
+
+module Rubycritic
+  module Platforms
+    class Base
+      attr_reader :browser_name
+
+      def initialize
+        @browser_name = Config.open_with || ""
+      end
+
+      def self.can_open?
+        raise NotImplementedError
+      end
+
+      def browser
+        raise NotImplementedError
+      end
+
+      def open(_)
+        raise NotImplementedError
+      end
+
+      def application
+        raise NotImplementedError
+      end
+
+      def browser_hash
+        raise NotImplementedError
+      end
+
+      def platform
+        @platform ||= begin
+          if RUBY_PLATFORM =~ /java/
+            `uname -s`.downcase
```

Comment:
Wouldn't this crash for somebody running JRuby on windows?
How about doing something like [here](http://stackoverflow.com/questions/11784109/detecting-operating-systems-in-ruby)?